src/app/modules/shared/components/details-fields/details-fields.component.html

```diff
@@ -93,7 +93,11 @@
             (ngModelChange)="onStartDateChange($event)"
             [max]="getCurrentDate()"
             onkeydown="return false"
+            (click)="datepickerStartDate.opened ? datepickerStartDate.close() : datepickerStartDate.open()"
```

Comment:
In this part, you could refactor to a function, since you use the same comparison as above in line 129 

```html
...
<input
            matInput
            formControlName="start_date"
            id="start_date"
            class="form-control"
            aria-label="Small"
            aria-describedby="inputGroup-sizing-sm"
            (ngModelChange)="onStartDateChange($event)"
            [max]="getCurrentDate()"
            onkeydown="return false"
            (click)="openOrCloseDatePicker(datepickerStartDate)"
            ...
```
...
```javascript
openOrCloseDatePicker(datepicker: YOUR_TYPE_HERE) {
   return datepicker.opened ? datepicker.close() : datepicker.open();
}
```